File: Python version/specializations/dlv/desktop/dlv_desktop_service.py
from platforms.desktop.desktop_service import DesktopService
from specializations.dlv.dlv_answer_sets import DLVAnswerSets


class DLVDesktopService(DesktopService):
    """Extention of DesktopService for DLV."""

    def __init__(self, exe_path):
        super(DLVDesktopService, self).__init__(exe_path)
        self._load_from_stdin_option = "--"

    def _get_output(self, output, error):
        """Returns a new DLVAnwerSets from given output and error."""
        return DLVAnswerSets(output, error)

File: CSharp version/EmbASP/it/unical/mat/embasp/specializations/dlv/desktop/DLVDesktopService.cs
namespace it.unical.mat.embasp.specializations.dlv.desktop
{
	using Output = it.unical.mat.embasp.@base.Output;
	using DesktopService = it.unical.mat.embasp.platforms.desktop.DesktopService;

	public class DLVDesktopService : DesktopService
	{

		public DLVDesktopService(string exe_path) : base(exe_path)
		{
			load_from_STDIN_option = "--";
		}

		protected internal override Output getOutput(string output, string error)
		{
			return new DLVAnswerSets(output, error);
		}

	}

}